modules/docker/stop_test.go

```diff
@@ -0,0 +1,30 @@
+package docker
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestStop(t *testing.T) {
+	t.Parallel()
+
+	// appending timestamp to container name to run tests in parallel
+	name := "test-nginx" + strconv.FormatInt(time.Now().UnixNano(), 10)
+
+	// for testing the stopping of a docker container
+	// we got to run a container first and then stop it
+	runOpts := &RunOptions{
+		Detach: true,
+		Name:   name,
+		Remove: true,
+	}
+	Run(t, "nginx:1.17-alpine", runOpts)
+
+	// try to stop it now
+	stopOpts := &StopOptions{}
+	out := Stop(t, []string{name}, stopOpts)
+	require.Contains(t, out, name)
```

Comment:
NIT: Would be nice to enhance the test with a check to see if the container actually stopped. A few options for this:

- Use `shell` to call `docker ps` to see if this container stopped.
- Since we are using nginx, we can use `HttpGet` to see if the container is unreachable. This requires an equivalent `HttpGet` check after the `Run` call to make sure the container is up.

Reply:
@yorinasub17 I would prefer to use checks that are independent of other tests, `docker ps` output may vary with other running tests, so I implemented by calling out nginx using `HttpGet`, thanks for suggesting.

Reply:
Updated gist with test run: https://gist.github.com/dhwaneetbhatt/018f0775493143b30b7b463deaa45e37

Reply:
Thanks for adding the http get calls. However, I am worried about the brittleness of this test, especially under the load of all the other tests. E.g I am pretty sure there is a possibility that there is a delay between `docker run` and the nginx server being available, and so there might be a race condition between the `HttpGet` call and nginx coming up.

Can you update those checks to instead use https://github.com/gruntwork-io/terratest/blob/master/modules/http-helper/http_helper.go#L120 for the confirmation check (wait until nginx comes up), and https://github.com/gruntwork-io/terratest/blob/master/modules/retry/retry.go#L53 for the negation check (wait until nginx comes down)? 2 minutes is probably enough for the wait (60 for max retries, 2*time.Second for sleep between retries).

Let me know if it isn't clear how to use those functions and happy to provide additional pointers.

Reply:
@yorinasub17 made the changes. For the nginx going down, the `docker stop` command is actually synchronous and will wait for container to die before returning. So, we can just verify by executing `docker ps` along with name filter and ensure it does not return any values.

Made another minor change regarding godoc comment in `run.go`, and added `.vscode` in .gitignore since I use vscode.

Let me know if it looks good.